p2p/src/main/java/bisq/network/p2p/peers/getdata/RequestDataHandler.java

```diff
@@ -304,6 +278,56 @@ public void stop() {
     // Private
     ///////////////////////////////////////////////////////////////////////////////////////////
 
+    private void cleanupRecentlyRemovedCache() {
+        while (recentlyRemovedMessages.size() > MAX_CACHE_SIZE)
+            recentlyRemovedMessages.remove(0);
+    }
+
+    private void logContents(NetworkEnvelope networkEnvelope,
+                             Set<ProtectedStorageEntry> dataSet,
+                             Set<PersistableNetworkPayload> persistableNetworkPayloadSet) {
+        Map<String, Set<NetworkPayload>> payloadByClassName = new HashMap<>();
+        dataSet.stream().forEach(e -> {
+            final ProtectedStoragePayload protectedStoragePayload = e.getProtectedStoragePayload();
+            if (protectedStoragePayload == null) {
+                log.warn("StoragePayload was null: {}", networkEnvelope.toString());
+                return;
+            }
+
+            // For logging different data types
+            String className = protectedStoragePayload.getClass().getSimpleName();
+            if (!payloadByClassName.containsKey(className))
+                payloadByClassName.put(className, new HashSet<>());
+
+            payloadByClassName.get(className).add(protectedStoragePayload);
+        });
+
+
+        if (persistableNetworkPayloadSet != null) {
+            persistableNetworkPayloadSet.stream().forEach(persistableNetworkPayload -> {
+                // For logging different data types
+                String className = persistableNetworkPayload.getClass().getSimpleName();
+                if (!payloadByClassName.containsKey(className))
+                    payloadByClassName.put(className, new HashSet<>());
+
+                payloadByClassName.get(className).add(persistableNetworkPayload);
+            });
+        }
+
+        // Log different data types
+        StringBuilder sb = new StringBuilder();
+        sb.append("\n#################################################################\n");
+        sb.append("Connected to node: " + peersNodeAddress.getFullAddress() + "\n");
+        final int items = dataSet.size() +
+                (persistableNetworkPayloadSet != null ? persistableNetworkPayloadSet.size() : 0);
+        sb.append("Received ").append(items).append(" instances\n");
+        payloadByClassName.entrySet().stream().forEach(e -> sb.append(e.getKey())
+                .append(": ")
+                .append(e.getValue().size())
+                .append("\n"));
+        sb.append("#################################################################");
+        log.info(sb.toString());
```

Comment:
info is probably enabled most of the time. still this whole method should be in a `if (log.isInfoEnabled)` block